tests/suites/coslite/cl.sh

```diff
@@ -0,0 +1,62 @@
+# Ensure that COS Lite deploys successfully, the
+# relations are setup as expected and we can access the dashboards.
+run_deploy_coslite() {
+	echo
+
+	local model_name file overlay_path admin_passwd alertmanager_ip grafana_ip prometheus_ip
+	model_name="deploy-coslite"
+	file="${TEST_DIR}/${model_name}.log"
+	ensure "${model_name}" "${file}"
+
+	overlay_path="./tests/suites/coslite/overlay"
+	juju deploy cos-lite --trust --channel=beta --overlay "${overlay_path}/offers-overlay.yaml" --overlay "${overlay_path}/storage-small-overlay.yaml"
+
+	echo "Wait for all unit agents to be in active condition"
+	wait_for 0 '[.applications[] | select((.units[] | .["juju-status"].current != "idle") and (.units[] | .["workload-status"].current != "error"))] | length' 1800
+
+	echo "Check that all offer endpoints specified in the overlays exist"
+	wait_for 5 '[.offers[] | .endpoints] | length'
+
+	admin_passwd=$(juju run-action --wait grafana/0 get-admin-password --format json | jq '.["unit-grafana-0"]["results"]["admin-password"]')
+	if [ -z "$admin_passwd" ]; then
+		echo "expected to get admin password for grafana/0"
+		exit 1
+	fi
+
+	# Assert the web dashboards are reachable
+	alertmanager_ip=$(juju status --format json | jq '.applications["alertmanager"]["units"]["alertmanager/0"].address' | tr -d '"')
+	echo "Check if alertmanager is ready to serve traffic"
+	curl -sS http://"$alertmanager_ip":9093/ready -o /dev/null
+	grafana_ip=$(juju status --format json | jq '.applications["grafana"]["units"]["grafana/0"].address' | tr -d '"')
+	echo "Check if grafana is ready to serve traffic"
+	curl -sS http://"$grafana_ip":3000/api/health -o /dev/null
+	prometheus_ip=$(juju status --format json | jq '.applications["prometheus"]["units"]["prometheus/0"].address' | tr -d '"')
+	echo "check if prometheus is ready to serve traffic"
+	curl -sS http://"$prometheus_ip":9090/ready -o /dev/null
+	echo "cos lite tests passed"
+
+	# Cleanup
+	echo "Removing applications and their attached storage volume"
+	juju remove-application alertmanager --destroy-storage
+	# without --force grafana get stuck in a hook(removal) error state.
+	juju remove-application grafana --destroy-storage --force
+	juju remove-application loki --destroy-storage
+	juju remove-application prometheus --destroy-storage
```

Comment:
You might only need to remove grafana here. And you may also not need to include `--destroy-storage` since (for me) that part works fine within destroy_model. Only explicitly removing grafana will also make it clearer you're only doing this out of necessity so worth trying